db/src/events.rs

```diff
@@ -35,6 +35,23 @@ pub struct DbEvent {
 pub type DbEventSender = broadcast::Sender<DbEvent>;
 pub type DbEventReceiver = broadcast::Receiver<DbEvent>;
 
+pub struct NotificationListeningOpts {
+    channel_name: String,
+}
+
+impl NotificationListeningOpts {
+    pub fn on_table(table_name: &str) -> Self {
```

Comment:
I think this should be typed, the only necessary addition is to add `#[derive(AsRefStr)]` and `#[strum(serialize_all = "snake_case")]` to `DbTable`.

```suggestion
    pub fn on_table(table_name: DbTable) -> Self {
```